apps/src/blockly/addons/cdoXml.js

```diff
@@ -148,26 +148,76 @@ export function addMutationToBehaviorDefBlocks(blockElement) {
   }
 }
 
+/**
+ * Adds a mutation element to a block if it's a text join block with an input count.
+ * CDO Blockly uses an unsupported method for serializing input count state
+ * where an arbitrary block attribute could be used to manage extra state.
+ * Mainline Blockly expects a mutator. The presence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ */
+export function addMutationToTextJoinBlocks(blockElement) {
+  if (
+    !['text_join', 'text_join_simple'].includes(
+      blockElement.getAttribute('type')
+    )
+  ) {
+    return;
+  }
+  const mutationElement =
+    blockElement.querySelector('mutation') ||
+    blockElement.ownerDocument.createElement('mutation');
+  // Place mutator before fields, values, and other nested blocks.
+  blockElement.insertBefore(mutationElement, blockElement.firstChild);
+
+  // We need to keep track of the expected number of inputs in order to create them all.
+  // If not, it needs a static behavior id in order to be translatable
```

Comment:
Are the comments about behavior id and `usercreated` relevant here? I don't see any logic relating to those in this function.

Reply:
Not relevant.  They should have been deleted. Will remove!